kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -967,6 +984,37 @@ func (c *ipamController) nodeExists(knode string) bool {
 	return true
 }
 
+// nodeIsBeingMigrated looks up a node and checks, if it is marked by the flannel-migration controller to undergo migration.
+func (c *ipamController) nodeIsBeingMigrated(name string) (bool, error) {
+	// Get node to inspect labels
+	obj, ok, err := c.nodeIndexer.GetByKey(name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to check node for migration status: %w", err)
+	}
+	if !ok {
```

Comment:
I believe this is the `IsNotFound` case - we probably want to structure this like so:

```go
obj, ok, err := c.nodeIndexer.GetByKey(name)
if !ok {
  // Node doesn't exist, so isn't being migrated.
  return false, nil
}
if err != nil {
  return false, err
}
...